types/govuk-frontend/dist/govuk/common/index.d.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * Common helpers which do not require polyfill.
+ *
+ * IMPORTANT: If a helper require a polyfill, please isolate it in its own module
+ * so that the polyfill can be properly tree-shaken and does not burden
+ * the components that do not need that helper
+ */
+/**
+ * Config merging function
+ *
+ * Takes any number of objects and combines them together, with
+ * greatest priority on the LAST item passed in.
+ *
+ * @internal
+ * @param {...{ [key: string]: unknown }} configObjects - Config objects to merge
+ * @returns {{ [key: string]: unknown }} A merged config object
+ */
+export function mergeConfigs(
+    ...configObjects: Array<{
+        [key: string]: unknown;
+    }>
+): {
+    [key: string]: unknown;
+};
+
+/**
+ * Extracts keys starting with a particular namespace from dataset ('data-*')
+ * object, removing the namespace in the process, normalising all values
+ *
+ * @internal
+ * @param {{ schema: Schema }} Component - Component class
```

Comment:
it's minor, but .d.ts jsdoc shouldn't have types since they're redundant with the types which are actually checked -- which means that it's easier for them to fall out of date in addition.

Reply:
Thanks @sandersn, it's a shame the compiler leaves these in

Alternatives were [`removeComments`](https://www.typescriptlang.org/tsconfig/#removeComments) (all comments removed) or edit all JSDoc by hand

The output is from source JavaScript via `tsc --build` so they'll always match the types, e.g.

https://github.com/alphagov/govuk-frontend/blob/953f15903426a8dc508592ed7dad90a253b058d4/packages/govuk-frontend/src/govuk/common/index.mjs#L50-L53